### PR TITLE
CHET-173: Refactor S3FilesystemMigrationService to be more testable, and expand integration tests

### PR DESCRIPTION
**Pull request by tarka**
This is separate from the service refactoring; this one ensures that we have a working baseline for the S3FilesystemMigrationService tests.

### CODE REVIEW

File: src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java [hunk 1]
```diff
@@ -19,54 +18,90 @@
...
-    private SchedulerService schedulerService;
+    AtlassianPluginAWSCredentialsProvider credentialsProvider;
 
     @BeforeEach
     void setup() throws Exception {
+        credentialsProvider = new AtlassianPluginAWSCredentialsProvider(new AwsCredentialsProviderShim(s3.getDefaultCredentialsProvider()));
+
+        s3AsyncClient = S3AsyncClient.builder()
+            .endpointOverride(new URI(s3.getEndpointConfiguration(S3).getServiceEndpoint()))
+            .credentialsProvider(new AwsCredentialsProviderShim(s3.getDefaultCredentialsProvider()))
```
Comment (theghostwhoforks) [inline src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java]: why do we need the shim? As opposed to using the defaultCredentialsProvider from S3?

File: src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java [hunk 2]
```diff
@@ -19,54 +18,90 @@
...
         fsService.startMigration();
 
         Assertions.assertNotEquals(FilesystemMigrationStatus.FAILED, fsService.getReport().getStatus());
+
+        HeadObjectRequest req = HeadObjectRequest.builder()
+            .bucket(bucket)
+            .key(file.getFileName().toString())
+            .build();
+        HeadObjectResponse resp = s3AsyncClient.headObject(req).get();
+        assertTrue(resp.sdkHttpResponse().isSuccessful());
```
Comment (theghostwhoforks) [inline src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java]: Additionally, do we need to assert for content or size?

File: src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java (hunk 1 above)
Comment (tarka) [reply src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java]: TestContainers/LocalStack provides v1 credentials, but we use the v2 API. It's just a quick wrapper around converting them.

File: src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java (hunk 2 above)
Comment (tarka) [reply src/test/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationServiceIT.java]: Ideally, but I'm being slack ;) Thats a bit more work and I need a baseline. I'll add some more test utils in the refactor.

**Pull request merged**